plugins/source/aws/codegen/recipes/base.go

```diff
@@ -418,3 +425,42 @@ func CreateReplaceTransformer(replace map[string]string) func(field reflect.Stru
 		return name, nil
 	}
 }
+
+func validateServiceMultiplex(multiplexerCall string) error {
+	re := regexp.MustCompile(`\"(.*?)\"`)
+	// Find the value of the service parameter
+	submatchAll := re.FindStringSubmatch(multiplexerCall)
+	t := client.ReadSupportedServiceRegions()
+
+	var services []string
+	for _, partition := range t.Partitions {
+		for service := range partition.Services {
+			services = appendIfMissing(services, service)
```

Comment:
Does using a map here make it more complex? Could then use some `funk` functs and get rid of the helpers?

Reply:
Good call! Didn't even need the `funk`... In the original implementation I had tried to use the generic functions in `go/map` and `go/slice` but in the end it devolved into recreating set functionality

Reply:
fixed